packages/db/tests/lib/test-QuerySearchClient.js

```diff
@@ -0,0 +1,256 @@
+      granule_id: testGranule.granule_id,
+    }
+  );
+  t.like(
+    await querySearchClient.shift(),
+    {
+      ...records[1],
+      granule_cumulus_id: Number.parseInt(records[1].granule_cumulus_id, 10),
+      granule_id: testGranule.granule_id,
+    }
+  );
+});
+
+test('QuerySearchClient.shift() returns next record if next record must be fetched', async (t) => {
+  const { knex, bucket, testGranule } = t.context;
+
+  const records = orderBy(await createFileRecords(t.context, 2), ['cumulus_id']);
+
+  const query = getFilesAndGranuleInfoQuery({
+    knex,
+    searchParams: { bucket },
+    sortColumns: ['cumulus_id'],
+    granuleColumns: ['granule_id'],
+  });
+  const querySearchClient = new QuerySearchClient(
+    query,
+    1
+  );
+  const queryOffsetSpy = sinon.spy(query, 'offset');
+  const queryLimitSpy = sinon.spy(query, 'limit');
+
+  t.like(
+    await querySearchClient.shift(),
+    {
+      ...records[0],
+      granule_cumulus_id: Number.parseInt(records[0].granule_cumulus_id, 10),
+      granule_id: testGranule.granule_id,
+    }
+  );
+  t.like(
+    await querySearchClient.shift(),
+    {
+      ...records[1],
+      granule_cumulus_id: Number.parseInt(records[1].granule_cumulus_id, 10),
+      granule_id: testGranule.granule_id,
+    }
+  );
+  t.is(queryOffsetSpy.callCount, 2);
+  t.is(queryLimitSpy.callCount, 2);
+});
+
+test('QuerySearchClient.peek() correctly returns true if next record exists in fetched results', async (t) => {
+  const { knex, bucket } = t.context;
+
+  await createFileRecords(t.context, 1);
+
+  const query = getFilesAndGranuleInfoQuery({
+    knex,
+    searchParams: { bucket },
+    sortColumns: ['cumulus_id'],
+    granuleColumns: ['granule_id'],
+  });
+  const fileSearchClient = new QuerySearchClient(
+    query,
+    1
+  );
+  t.truthy(
+    await fileSearchClient.peek()
+  );
+});
+
+test('QuerySearchClient.peek() correctly returns true if next record must be fetched', async (t) => {
+  const { knex, bucket } = t.context;
+
+  await createFileRecords(t.context, 2);
+
+  const query = getFilesAndGranuleInfoQuery({
+    knex,
+    searchParams: { bucket },
+    sortColumns: ['cumulus_id'],
+    granuleColumns: ['granule_id'],
+  });
+  const queryOffsetSpy = sinon.spy(query, 'offset');
+  const queryLimitSpy = sinon.spy(query, 'limit');
+  const fileSearchClient = new QuerySearchClient(
+    query,
+    1
+  );
+
+  t.truthy(
+    await fileSearchClient.peek()
+  );
+  await fileSearchClient.shift();
+  t.truthy(
+    await fileSearchClient.peek()
+  );
+  t.is(queryOffsetSpy.callCount, 2);
+  t.is(queryOffsetSpy.getCall(0).args[0], 0);
+  t.is(queryOffsetSpy.getCall(1).args[0], 1);
+  t.is(queryLimitSpy.callCount, 2);
+  t.is(queryLimitSpy.getCall(0).args[0], 1);
+  t.is(queryLimitSpy.getCall(1).args[0], 1);
+});
+
+test('QuerySearchClient.peek() correctly returns false if next record does not exist in fetched results', async (t) => {
```

Comment:
Consider updating this to something like:
```suggestion
test('QuerySearchClient.peek() correctly returns undefined if next record does not exist in fetched results', async (t) => {
```